types/module/simulation.go

```diff
@@ -115,16 +115,16 @@ func (sm *SimulationManager) WeightedOperations(simState SimulationState) []simu
 // SimulationState is the input parameters used on each of the module's randomized
 // GenesisState generator function
 type SimulationState struct {
-	AppParams    simulation.AppParams
-	Cdc          codec.JSONCodec                      // application codec
-	Rand         *rand.Rand                           // random number
-	GenState     map[string]json.RawMessage           // genesis state
-	Accounts     []simulation.Account                 // simulation accounts
-	InitialStake sdkmath.Int                          // initial coins per account
-	NumBonded    int64                                // number of initially bonded accounts
-	BondDenom    string                               // denom to be used as default
-	GenTimestamp time.Time                            // genesis timestamp
-	UnbondTime   time.Duration                        // staking unbond time stored to use it as the slashing maximum evidence duration
-	ParamChanges []simulation.ParamChange             // simulated parameter changes from modules
-	Contents     []simulation.WeightedProposalContent // proposal content generator functions with their default weight and app sim key
+	AppParams         simulation.AppParams
+	Cdc               codec.JSONCodec                      // application codec
+	Rand              *rand.Rand                           // random number
+	GenState          map[string]json.RawMessage           // genesis state
+	Accounts          []simulation.Account                 // simulation accounts
+	InitialStake      sdkmath.Int                          // initial coins per account
+	NumBonded         int64                                // number of initially bonded accounts
+	BondDenom         string                               // denom to be used as default
+	GenTimestamp      time.Time                            // genesis timestamp
+	UnbondTime        time.Duration                        // staking unbond time stored to use it as the slashing maximum evidence duration
+	LegacyParamChange []simulation.LegacyParamChange       // simulated parameter changes from modules
```

Comment:
Why do we still need the legacy ones here? Param changes are now just messages, so why keep the legacy field here?

Reply:
Not everyone will have migrated away from `x/params`, so I believe if other modules uses SDK simulations but haven't migrated yet, they will use this. I renamed it to make clear it's a _legacy_ way.

Reply:
Fair point 👍